content/anchor-frontend.md

```diff
@@ -0,0 +1,619 @@
+```
+
+## Anchor `MethodsBuilder`
+
+Once the `Program` object is set up, we can use the Anchor `MethodsBuilder` to build transactions with instructions from our program. The `MethodsBuilder` uses the `IDL` to provides a simplified format building transaction to invoke program instructions.
+
+The basic `MethodsBuilder` format looks like this:
+
+```tsx
+// sends transaction
+await program.methods
+  .instructionName(instructionDataInputs)
+  .accounts({})
+  .signers([])
+  .rpc()
+```
+
+The basic format includes the following:
+
+- `program` - the program being invoked specified by the `programId` from the `Program` object
+- `methods` -  builder API for all APIs on the program and includes all instructions from the `IDL`
+- `instructionName` - the name of the specific instruction from the `IDL` to invoke
+- `instructionDataInputs` - include any instruction data required by the instruction within the parentheses after the instruction name
+- `accounts`- list of accounts required by the instruction being invoked
+- `signers` - any additional signers required by the instruction
+- `rpc` -  creates and sends a signed transaction with the specified instruction and returns a `TransactionSignature`. When using `.rpc`, the `Wallet` from the `Provider` is automatically included as a signer and does not have to be listed explicitly stated.
+
+Note that if no additional signers are required by the instruction other than the `Wallet` specified with the `Provider`, the `.signer([])` line can be excluded.
+
+You can also build the transaction directly by changing `.rpc()` to `.transaction()`.  This builds a `Transaction` object using the instruction specified.
+
+```tsx
+// creates transaction
+const transaction = await program.methods
+  .instructionName(instructionDataInputs)
+  .accounts({})
+  .transaction()
+
+await sendTransaction(transaction, connection)
+```
+
+Similarly, you can use the same format to build an instruction using `.instruction` and then manually add the instructions to a new transaction. This builds a `TransactionInstruction` object using the instruction specified.
+
+```tsx
+// creates first instruction
+const instructionOne = await program.methods
+  .instructionOneName(instructionOneDataInputs)
+  .accounts({})
+  .instruction()
+
+// creates second instruction
+const instructionTwo = await program.methods
+  .instructionTwoName(instructionTwoDataInputs)
+  .accounts({})
+  .instruction()
+
+// add both instruction to one transaction
+const transaction = new Transaction().add(instructionOne, instructionTwo)
+
+// send transaction
+await sendTransaction(transaction, connection)
+```
+
+In summary, the Anchor `MethodsBuilder` provides a simplified and more flexible way to interact with on-chain programs. You can build an instruction, a transaction, or build and send a transaction using basically the same format without having to manually serialize or deserialize the accounts or instruction data.
```

Comment:
love this section

I think we should also mention the differences in naming conventions when anchor makes an idl (maybe this is more for the previous lesson) i.e. rust -> instruction_number_one, IDL -> instructionNumberOne

Reply:
Added sentence about naming convention at beginning of section

> ## Anchor `MethodsBuilder`
> 
> Once the `Program` object is set up, we can use the Anchor `MethodsBuilder` to build transactions with instructions from our program. The `MethodsBuilder` uses the `IDL` to provide a simplified format for building transactions to invoke program instructions. Note that the camel case naming convention is used when interacting with a program from the client, compared to the snake case naming convention used when the writing the program in rust.